js/node_shared/unix.js

```diff
@@ -0,0 +1,369 @@
+    if (!children) {
+      throw new Error(`unix_waitpid: no children process of parent {pid_req}`);
+    }
+    for (const child of children) {
+      return handleChild(child);
+    }
+    // specific process
+  } else if (pid_req > 0) {
+    let child = globalThis.pidToProcessTable.get(pid_req);
+    if (!child) {
+      throw new Error(`unix_waitpid: no child at pid {pid_req}`);
+    }
+
+    return handleChild(child);
+    // process group nonsense
+  } else {
+    function getAllDescendants(pid) {
+      let children = globalThis.pidToChildrenTable.get(pid);
+      if (!children) {
+        return [pid];
+      }
+
+      let acc = [];
+      for (const child of children) {
+        acc = acc.concat(getAllDescendants(child.pid));
+      }
+      return [pid].concat(acc);
+    }
+
+    let descendants = getAllDescendants(process.pid);
+    for (const child of descendants) {
+      return handleChild(child);
+    }
+  }
+}
+
+//Provides: unix_pipe
+function unix_pipe(cloexec, vunit) {
+  console.error("unix_pipe");
+  console.error(cloexec);
+  console.error(vunit);
+
+  // The pipe we spawn here will eventually reach unix_spawn.
+  // We don't want to use a real file descriptor, bceause we're trying
+  // to avoid them, in favor of Node's redirection ability, which will open
+  // the corresponding file descriptors for us.
+  // Because this call requires us to decide on what the file descriptor is
+  // ahead of time, though, we will just pass around a simulated FD, which
+  // in conjunction with fdTable, will let us recover the content which is
+  // supposed to be written to that FD.
+  let fd = globalThis.fdCount;
+
+  globalThis.fdCount++;
+
+  return [0, fd, fd];
+}
+
+//Provides: unix_set_close_on_exec
+function unix_set_close_on_exec(fd) {
+  console.error("unix_set_close_on_exec");
+  console.error(fd);
+  return;
+}
+
+//Provides: unix_getcwd
+//Requires: caml_sys_getcwd
+function unix_getcwd(vunit) {
+  console.error("unix_cwd");
+
+  return caml_sys_getcwd(vunit);
+}
+
+//Provides: unix_chdir
+//Requires: caml_sys_chdir
+function unix_chdir(path) {
+  console.error("unix_chdir");
+  console.error(path);
+
+  caml_sys_chdir(path);
+
+  return;
+}
+
+//Provides: unix_clear_close_on_exec
+function unix_clear_close_on_exec(fd) {
+  console.error("unix_clear_close_on_exec");
+  console.error(fd);
+  return;
+}
+
+//Provides: unix_dup
+//Requires: unix_pipe
+function unix_dup(cloexec, fd) {
+  console.error("unix_dup");
+  console.error(cloexec);
+  console.error(fd);
+
+  // We just use the same unix_pipe logic to generate a fresh fake
+  // file descriptor.
+  return unix_pipe(cloexec, fd);
+}
+
+//Provides: unix_close
+function unix_close(fd) {
+  console.error("unix_close");
+  console.error(fd);
+
+  return;
+}
+
+//Provides: unix_fork
+function unix_fork(vunit) {
+  console.error("unix_fork");
+  console.error(vunit);
+
+  return;
+}
+
+//Provides: unix_setitimer
+function unix_setitimer(it, its) {
```

Comment:
maybe not in this PR, but we should either make sure scan timeout is set to 0 so this code never runs, or we should properly handle it. I'm fine with no file timeouts for a beta, but it's something we should def do before it's the default